Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type or provide a default value, which allows to infer the type");
```

Comment:
As a user though, I would not expect changing my default value from this
```swift
class Foo: RLMObject {
    var bar: NSNumber = 0.5
}
```
to this
```swift
class Foo: RLMObject {
    var bar: NSNumber = 1
}
```
to require a schema migration. That would be extremely surprising.